VisualPinball.Unity/VisualPinball.Unity/Import/VpxImporter.cs

```diff
@@ -154,34 +155,42 @@ private void ImportRenderObjects(IRenderable item, RenderObjectGroup rog, GameOb
-
 			} else if (rog.HasChildren) {
 				foreach (var ro in rog.RenderObjects) {
 					var subObj = new GameObject(ro.Name);
 					subObj.transform.parent = obj.transform;
+					subObj.layer = ChildObjectsLayer;
 					ImportRenderObject(item, ro, subObj);
 				}
 			}
 
 			// apply transformation
 			if (rog.HasChildren) {
-				SetTransform(obj.transform, rog.TransformationMatrix.ToUnityMatrix());
+				obj.transform.SetFromMatrix(rog.TransformationMatrix.ToUnityMatrix());
 			}
 
 			// add unity component
+			MonoBehaviour ic = null;
 			switch (item) {
-				case Bumper bumper: obj.AddComponent<BumperBehavior>().SetData(bumper.Data); break;
-				case Flipper flipper: flipper.SetupGameObject(obj, rog); break;
-				case Gate gate: obj.AddComponent<GateBehavior>().SetData(gate.Data); break;
-				case HitTarget hitTarget: obj.AddComponent<HitTargetBehavior>().SetData(hitTarget.Data); break;
-				case Kicker kicker: obj.AddComponent<KickerBehavior>().SetData(kicker.Data); break;
-				case Primitive primitive: obj.AddComponent<PrimitiveBehavior>().SetData(primitive.Data); break;
-				case Ramp ramp: obj.AddComponent<RampBehavior>().SetData(ramp.Data); break;
-				case Rubber rubber: obj.AddComponent<RubberBehavior>().SetData(rubber.Data); break;
-				case Spinner spinner: obj.AddComponent<SpinnerBehavior>().SetData(spinner.Data); break;
-				case Surface surface: surface.SetupGameObject(obj, rog); break;
-				case Table table: table.SetupGameObject(obj, rog); break;
-				case Trigger trigger: obj.AddComponent<TriggerBehavior>().SetData(trigger.Data); break;
+				case Bumper bumper:			ic = obj.AddComponent<BumperBehavior>().SetData(bumper.Data); break;
+				case Flipper flipper:		ic = flipper.SetupGameObject(obj, rog); break;
+				case Gate gate:				ic = obj.AddComponent<GateBehavior>().SetData(gate.Data); break;
+				case HitTarget hitTarget:	ic = obj.AddComponent<HitTargetBehavior>().SetData(hitTarget.Data); break;
+				case Kicker kicker:			ic = obj.AddComponent<KickerBehavior>().SetData(kicker.Data); break;
+				case Primitive primitive:	ic = obj.AddComponent<PrimitiveBehavior>().SetData(primitive.Data); break;
+				case Ramp ramp:				ic = obj.AddComponent<RampBehavior>().SetData(ramp.Data); break;
+				case Rubber rubber:			ic = obj.AddComponent<RubberBehavior>().SetData(rubber.Data); break;
+				case Spinner spinner:		ic = obj.AddComponent<SpinnerBehavior>().SetData(spinner.Data); break;
+				case Surface surface:		ic = surface.SetupGameObject(obj, rog); break;
+				case Table table:			ic = table.SetupGameObject(obj, rog); break;
+				case Trigger trigger:		ic = obj.AddComponent<TriggerBehavior>().SetData(trigger.Data); break;
+			}
+			// for convenience move item behavior to the top of the list
+			if (ic != null) {
+				int numComp = obj.GetComponents<MonoBehaviour>().Length;
+				for (int i = 0; i <= numComp; i++) {
+					UnityEditorInternal.ComponentUtility.MoveComponentUp(ic);
+				}
 			}

```

Comment:
Does this work? Is that the order of the `MonoBehaviors` in the inspector?

![image](https://user-images.githubusercontent.com/70426/83459779-5a307a00-a465-11ea-81cc-76a76fde77af.png)

Reply:
Hmm it seemed to work for me, its certainly not a supported API, but its the only way I could find to do it progammatically. Do you have a VPX file that repros this I could test?